evaluation.py

```diff
@@ -1138,28 +1144,112 @@ def run_ancestor_comparison(args):
+        df_all['width'] = np.repeat(sum_per_timeslice, sum_per_timeslice)
+        
+        mean_by_anc_time = df.iloc[df['nsites'].nonzero()].groupby('time', sort=True).mean()
+        median_by_anc_time = df.iloc[df['nsites'].nonzero()].groupby('time', sort=True).median()
+        sum_by_anc_time = df.iloc[df['nsites'].nonzero()].groupby('time', sort=True).sum()
+        
+        line_x = np.insert(np.cumsum(sum_by_anc_time['nsites']).values, 0, 0)
+        
+        if ancestors_are_estimated:
+            #averaging over times is probably more-or-less OK
+            lines_y = [mean_by_anc_time.l, median_by_anc_time.l]
+            names = ["Mean", "Median"]
+            linestyles = ["-",":"]
+        else:
+            #times are unique per ancestor, so we don't do well averaging - have to use a running mean
+            assert args.running_average_span % 2 == 1, "Must have odd number of bins"
+            lines_y = [
+                np.pad(
+                    running_mean(mean_by_anc_time.l.values, args.running_average_span), 
+                    (args.running_average_span-1)//2, mode='constant',constant_values=(np.nan,)),
+                np.pad(
+                    running_median(median_by_anc_time.l.values, args.running_average_span), 
+                    (args.running_average_span-1)//2, mode='constant',constant_values=(np.nan,))
+                ]
+            names = [
+                "Running mean over {} ancestors".format(args.running_average_span), 
+                "Running median over {} ancestors".format(args.running_average_span)]
+            linestyles = ["-",":"]
+            #save some stuff for when we plot inferred lines
+            exact_mean_line_y = lines_y[0]
+            exact_median_line_y = lines_y[1]
+            exact_line_x = line_x
+            max_y = np.max(df_all.lengths_per_site.values)
+        
+        fig = plt.figure(figsize=(10,10), dpi=100)
+        x_jittered = df_all.mean_x_pos.values + \
+            np.random.uniform(-df_all.width.values*9/20, df_all.width.values*9/20, len(df_all.mean_x_pos.values))
+        #plot with jitter
+        plt.scatter(x_jittered, df_all.lengths_per_site.values, marker='.', s=72./fig.dpi, alpha=0.75, color="black")
+        plt.ylim(1, max_y*1.02)
+        if args.log_yscale:
+            plt.yscale("log")
+        ax = plt.gca()
+        ax.set_xlim(xmin=0)
+        if ancestors_are_estimated:
+            plt.title("Ancestor lengths as estimated by tsinfer")
+            ax.step(exact_line_x[:-1], exact_mean_line_y, label="True mean", where='post', color="limegreen")
+            ax.step(exact_line_x[:-1], exact_median_line_y, label="True median", where='post', color="limegreen", linestyle=":")
+            plt.xlabel("Ancestors_freq (youngest to oldest)")
+            ax.set_xlim(xmin=0)
+            _ = ax.tick_params(axis='x', which="major", length=0)
```

Comment:
Why the ``_ = ax.tick_params(`` pattern here?

Reply:
When I run this in the interpreter, ax.tick_params() returns a huge list of the params used. Is there a better way to tell python not to print out the results of the call?

Reply:
Happy to remove. For testing, I usually run this in the interpreter, as it's hard to debug otherwise. Would appreciate some tips on how to do this better.

Reply:
OK, lets remove and merge this. Can talk about how to develop off line.